test/src/service/data_products/filtering_test.py

```diff
@@ -68,37 +77,79 @@ def test_to_arangosearch_aql():
         )),
     )
     for rf, expected in aql_set:
-        assert rf.to_arangosearch_aql("d.field", "pre") == expected
+        assert rf.to_arangosearch_aql("d.field", "pre", None) == expected
 
-def test_from_string_fail():
+
+def test_rangefilter_from_string_fail():
     test_set = (
-        (ColumnType.INT, "   \t   ", "^Missing range information$"),
+        (ColumnType.INT, "   \t   ", "Missing range information"),
         (
             ColumnType.INT,
             " [  \t   ",
-            "^Invalid range specification; expected exactly one comma:  \\[  \t   $"
+            "Invalid range specification; expected exactly one comma:  \\[  \t   "
```

Comment:
There are a bunch of  `\\` inside test_rangefilter_from_string_fail function. Why are they?

Reply:
Regex escape sequences

Reply:
Oh, I see. But why not using "\"? You also want to match a single "\" in your value as well?

Reply:
Python complains if you escape an invalid escape character:

```
  /home/crushingismybusiness/github/kbase/collections/test/src/service/data_products/filtering_test.py:89: DeprecationWarning: invalid escape sequence '\['
```